src/lib/pipeline.ml

```diff
@@ -777,4 +792,21 @@ module Compiler = struct
     in
     compiler.wrap_gtf_node pipeline gtf_node
 
-end
+  let hla_types_step ~compiler (pipeline : hla_types pipeline) =
+    let { machine ; work_dir; _ } = compiler in
+    match pipeline with
+    | Rna_hla_typer sample ->
+      match sample with
+      | Paired_end_sample (dataset, l1, l2) ->
+          (* TODO: Seq2HLA can actually take the gzipped version too, so we'd
+             need a unique type for that. *)
+          let r1 = fastq_step ~read:(`R1 dataset) ~compiler l1 in
+          let r2 = fastq_step ~read:(`R2 dataset) ~compiler l2 in
+          let work_dir = work_dir // ("seq2HLA_wd_" ^ dataset) in
+          Seq2HLA.hla_type ~host:Machine.(as_host machine) ~work_dir
+            ~run_with:machine ~run_name:dataset ~r1 ~r2
+      | _ -> failwithf
+              "Seq2HLA doesn't support single_end_samples or paired end \
+                samples of length greater than 1."
```

Comment:
right now `Paired_end_sample` will give you concatenated FASTQs so the length is already 1.